plugins/pluginmanagers/lz-n.nix

```diff
@@ -0,0 +1,130 @@
+{
+  lib,
+  helpers,
+  options,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+let
+  lzPluginType =
+    with helpers.nixvimTypes;
+    let
+      maybeRawStrOrList = oneOf [
+        str
+        (listOf (maybeRaw str))
+        rawLua
+      ];
+    in
+    submodule {
+      freeformType = attrsOf anything;
+      options = {
+        __unkeyed = mkOption {
+          type = str;
+          description = ''
+            The plugin's name (not the lua module name). This is what is passed to the load(name) function.
+            This is the name of the folder in /pack/*/opt/{name}.
+            See :h packadd
+          '';
+        };
+        enabled = helpers.defaultNullOpts.mkStrLuaFnOr bool true ''
+          When false, or if the function returns false, then this plugin will not be included in the spec.
+          This option corresponds to the `enabled` property of lz.n.
+        '';
+        beforeAll = helpers.mkNullOrLuaFn "Always executed before any plugins are loaded.";
+        before = helpers.mkNullOrLuaFn "Executed before this plugin is loaded.";
+        after = helpers.mkNullOrLuaFn "Executed after this plugin is loaded.";
+        event = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+        cmd = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on command.";
+        ft = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on filetype.";
```

Comment:
Also, remember that this is a freeform submodule: we don't need to have options for _all_ upstream options since they can be defined via the freeform type.

I don't mind having them, so long as we are confident they won't become a maintenance burden down the line. This happens when options are removed, renamed or re-typed upstream and we have to update accordingly.

The alternative is to document them as "examples" instead of as sub-options. I'm **not pushing** for you to change this, just prompting the discussion!

Reply:
(We could even go to the extreme of not having a submodlue and instead doing `maybeRaw (attrsOf anything)`)